tests/expectations/tests/issue-584-stylo-template-analysis-panic.rs

```diff
@@ -40,14 +40,11 @@ impl Clone for A {
     fn clone(&self) -> Self { *self }
 }
 #[repr(C)]
-#[derive(Debug, Copy, Clone)]
+#[derive(Debug, Default, Copy, Clone)]
 pub struct e<c> {
     pub d: RefPtr<c>,
     pub _phantom_0: ::std::marker::PhantomData<::std::cell::UnsafeCell<c>>,
 }
-impl <c> Default for e<c> {
-    fn default() -> Self { unsafe { ::std::mem::zeroed() } }
-}
```

Comment:
This expectation change seems to be from how if a type is hidden/blacklisted it will never end up inside this analysis's worklist because it isn't in the whitelisted items set, and therefore will never end up in the `cannot_derive_default` set. This means that our default is "can derive" for these types.

It isn't clear to me what the old default was; I think we would have looked inside the blacklisted item, which I'm not sure is what we want either. The type is blacklisted for some reason or another, and bindgen doesn't have the whole picture so how can it make an opinionated decision?

In light of that, I think the best thing to do would be to act conservatively, similar to how we do with blacklisted items in the template parameter analysis. With template params, we conservatively say that blacklisted items use all of their template parameters. In this case, conservative means "assume it cannot derive Default if it is blacklisted". As far as implementing this behavior goes, it means that we need to initially populate `cannot_derive_default` with all the items that are reachable from the whitelisted item set, but are not actually in the whitelisted item set (in other words, populate it with the blacklisted items).

@emilio wdyt? Does defaulting to "we should conservatively deny deriving default for blacklisted items" sound right to you?